### PR TITLE
LPS-129309 - Fix issues with tooltips that weren't working

### PR DESCRIPTION
**Pull request by bryceosterhaus**
https://issues.liferay.com/browse/LPS-129309

I finally think we have a conclusion here. We had to do a couple things to make this more consistent.

- We scope our tooltip provider to be within the `.lfr-tooltip-scope` class. Meaning if your element is under that selctor, title with automatically work with our tooltips.
- We now have a custom ReactPortal component, this helps us keep portals consistent and that they are all scoped for tooltips. We should enforce this via eslint as well, I will create an issue.
- We are now able to eliminate almost all uses of redundant TooltipProviders since we have a global one now. We may want an eslint rule for this as well to guide developers.

If I missed anything here, let me know.

cc @markocikos @matuzalemsteles

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
Labeling 3 squads since it sort of relates to all three

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
Well, I think we have come up with something more solid to implement Tooltip! @bryceosterhaus just one thing, since we need some part to stay under the scope of `.lfr-tooltip-scope`, by default will this work for applications that are using `render`? maybe we should add this within the scope by default so that all of our applications benefit from it? or is this being added elsewhere?

**Comment by bryceosterhaus** [comment]
@matuzalemsteles I am adding to our `render` method by default already, [here](https://github.com/liferay-frontend/liferay-portal/pull/1091/files#diff-fd718e2e69e6989357f190d3f023545d2b16f3f6655c31c70d67626aca40ce14R85). Is that what you were thinking?

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
> I am adding to our render method by default already, here. Is that what you were thinking?

Yeah, I think I'm blind, i hadn't seen it 🤦‍♂️.

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
@matuzalemsteles no worries, It's easily lost in all the file changes

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Looks fine to me! :)

File: modules/apps/frontend-js/frontend-js-tooltip-support-web/tsconfig.json
Comment (markocikos) [inline modules/apps/frontend-js/frontend-js-tooltip-support-web/tsconfig.json]: This is likely causing CI formatting fails. After https://github.com/liferay/liferay-frontend-projects/pull/550, our linter should no longer do this, right?

**Review by markocikos — APPROVED** [review]

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:stop

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
A story...

Screen Shot 2021-05-25 at 8 19 26 PM

Screen Shot 2021-05-25 at 8 19 56 PM

**Comment by matuzalemsteles** [comment]
I'll run the CI again, I think the tests are not related

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
Hey @nkkwon you could take a look if these tests are related I believe it is not but it would be good to check just to discard this, the tests are taking too long to run.

**Comment by nkkwon** [comment]
Just started reviewing :)

**Comment by nkkwon** [comment]
@bryceosterhaus @matuzalemsteles 
**Analysis details** 
It looks like these failures seem to be related to this PR

There are some semantic version increase required for
[semantic-versioning-jdk8/0/0
](https://testray.liferay.com/home/-/testray/case_results/1535778340?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D1535774702%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
```  
[exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.taglib.servlet                         MAJOR      2.0.0      2.1.0      3.0.0      VERSION INCREASE REQUIRED
```
[modules-compile-jdk8/0/0
](https://testray.liferay.com/home/-/testray/case_results/1535777666?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D1535774702%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
```  
 [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.portlet.expando.model.impl             MAJOR      1.3.0      1.4.0      2.0.0      VERSION INCREASE REQUIRED```

```    [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.portlet.expando.model.impl             MAJOR      1.3.0      1.4.0      2.0.0      VERSION INCREASE REQUIRED
  ```

Running these tests locally with the bundle also failed for me
[LocalFile.ContentPagesWithEditMode#CanOnlyEditContentInContentEditionMode
](https://testray.liferay.com/reports/production/logs/2021-05/test-1-1/test-portal-acceptance-pullrequest(master)/6669/functional-tomcat90-mysql57-jdk8/2/12/LocalFile.ContentPagesWithEditMode_CanOnlyEditContentInContentEditionMode/index.html.gz) - It looks like the test wasn't able to add the comment on the side bar

[LocalFile.Dropdown#DefineDisplayedBehaviorOfDropdownPanel 
](https://testray.liferay.com/reports/production/logs/2021-05/test-1-1/test-portal-acceptance-pullrequest(master)/6669/functional-tomcat90-mysql57-jdk8/3/6/LocalFile.Dropdown_DefineDisplayedBehaviorOfDropdownPanel/index.html.gz) - Looks like a flaky issue from https://issues.liferay.com/browse/LRQA-66718

[LocalFile.ExperienceCreateExperience#AddNewAnyoneExperience](https://testray.liferay.com/reports/production/logs/2021-05/test-1-1/test-portal-acceptance-pullrequest(master)/6669/functional-tomcat90-mysql57-jdk8/3/9/LocalFile.ExperienceCreateExperience_AddNewAnyoneExperience/index.html.gz) - Issue is the same as the one John explained in https://github.com/liferay-frontend/liferay-portal/pull/946#issuecomment-818050087

[LocalFile.MenuDisplay#DefineDisplayedSublevels](https://testray.liferay.com/reports/production/logs/2021-05/test-1-1/test-portal-acceptance-pullrequest(master)/6669/functional-tomcat90-mysql57-jdk8/7/10/LocalFile.MenuDisplay_DefineDisplayedSublevels/index.html.gz) - Looks to be a flaky test

**Comment by bryceosterhaus** [comment]
@nkkwon thanks for investigating! I will look into these and see what might be causing it

**Comment by bryceosterhaus** [comment]
> LocalFile.ContentPagesWithEditMode#CanOnlyEditContentInContentEditionMode

I wasn't able to reproduce this one locally with manual testing. Were you able to verify that the xpaths are still correct?

> LocalFile.ExperienceCreateExperience#AddNewAnyoneExperience

Also not able to reproduce this one locally with manual testing.


My guess is that the tests either need to update some xpath or need to wait shortly for the tooltip to appear. I believe our tooltips have a default delay of 600ms before appearing after a hover event.


cc @john-co did you ever hear back from poshi team on [this](https://github.com/liferay-frontend/liferay-portal/pull/946#issuecomment-818050087)?

**Comment by john-co** [comment]
Hi @bryceosterhaus, feedback was just confirming my suspicion to try some test fix to remove mouseOver from some click functions. There may be more test fixes needed to explore but I have most of the test fixes from previous PR in https://github.com/john-co/liferay-portal/pull/392/commits

**Comment by john-co** [comment]
Was ExperienceCreateExperience#AddNewAnyoneExperience the only test that failed? I remember a lot of tests failing in the first PR...will check if some other tests are included in this PR.

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
[Related]
- Content Page editor toolbar is left aligned
Experience_LeftAligned

**Comment by john-co** [comment]
ci:test:poshi

**Comment by john-co** [comment]
cc/ @david-gutierrez-mesa @Tim-Cao  I couldn't find any poshi test paths that click on the Content Page > Side bar > Mapping > Content Add button. Only macros/paths I found click on the select field instead of the add button mapEditableFieldToCollectionItems, gotoEditableFieldMapping. I don't see any other fixes related to Page Editor needed but let me know if there's something else I missed.

**Comment by bryceosterhaus** [comment]
@john-co I will fix the toolbar issue and re-run tests

**Comment by john-co** [comment]
I reproduced the LocalFile.ContentPagesWithEditMode#CanOnlyEditContentInContentEditionMode when running the poshi test locally. For some reason after clicking the heading fragment on the page, the sidebar text stays as `Comments` rather than a `Heading`...
```
Element is not present at "//h1[contains(@class,'page-editor__sidebar__panel-header')]//span[contains(.,'Heading')]"
```


However, when testing manually I cannot reproduce the issue. Will investigate further for possible fix on poshi side.
- Expected Behavior: After clicking on the heading fragment, the sidebar should show Heading
https://user-images.githubusercontent.com/35235266/120015821-a939be00-bf98-11eb-9239-d7555e605e9b.mp4

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by john-co** [comment]
For the failing test LocalFile.ContentPagesWithEditMode#CanOnlyEditContentInContentEditionMode it may be a mix of fragile test design and feature design imo but here's what I found:

- When editing a fragment the test relies on clicking on the page editor toolbar to exit the focus out of the fragment. 

https://user-images.githubusercontent.com/35235266/120023395-90360a80-bfa2-11eb-89e2-643b1d0910ac.mp4

- This PR affected this behavior so when clicking on the page editor toolbar, it does not exit the focus anymore. Thus causing a future step in the test to fail (Comment sidebar does not change to Heading)

https://user-images.githubusercontent.com/35235266/120023220-582ec780-bfa2-11eb-8c78-55bda90ba5d4.mp4

Is this behavior something you can fix so the test can pass @bryceosterhaus?

cc/ @david-gutierrez-mesa @Tim-Cao are there other ways to escape the focus of editing fragments? Or is this how we want to assert editing fragments to behave?

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
ci:test:stable

**Comment by bryceosterhaus** [comment]
ci:test:stable

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
every time CI passes

**Comment by matuzalemsteles** [comment]
Hey guys, is this on hold because we're waiting for the DXP release?

**Comment by markocikos** [comment]
@bryceosterhaus please fix conflict, `app-builder-web` got deleted.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles yeah still on hold until DXP release.

@markocikos I will push an update tomorrow without app-builder

**Comment by matuzalemsteles** [comment]
Thanks @bryceosterhaus for the info.

**Comment by bryceosterhaus** [comment]
I won't have a chance to fix this today, but GA2 still isn't released yet, so no rush since this won't be merged till after

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
@nkkwon can you verify test failures? Tests were passing before I rebased to fix merge conflicts.

**Comment by nkkwon** [comment]
Just started reviewing! :)

**Comment by nkkwon** [comment]
Analysis Details:
Looking through the failures on [testray](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=1540048414), it looks like these are just flaky errors.

Also looking at recent Acceptance analysis shows that Echo tests have been failing recently with these same errors.
None of these look unique to the PR to me, seems to be affecting multiple other PR's

**Comment by bryceosterhaus** [comment]
@nkkwon thanks! Going to manual forward to bchan, https://github.com/brianchandotcom/liferay-portal/pull/103175

**Pull request closed without merging**
